packages/shared/decorators/with-loader.tsx

```diff
@@ -0,0 +1,16 @@
+import * as React from "react";
+
+const withLoader = <T extends { [k: string]: any }>(
+  Component: React.ComponentType<T>
+): React.ComponentType<WithLoaderProps & T> => props => {
+  const { loader, condition = true } = props;
+  if (condition) return <Component {...props} />;
```

Comment:
We mustn't set `loader `and `condition `props to Component.
```suggestion
  if (condition) return <Component {...other as T} />;
```